### PR TITLE
Regenerate Client for LabApi using the latest labapi swagger spec 

### PR DESCRIPTION
**Pull request by iamgusain**
Regenerate Client for LabApi using the latest labapi swagger spec https://msidlab.com/swagger/v1/swagger.json
swagger codegen : https://swagger.io/tools/swagger-codegen/
Swagger codegen library version: [swagger-codegen-cli-3.0.25.jar](https://github.com/swagger-api/swagger-codegen)

Command used to generate client code: 
`swagger-codegen-cli-3.0.25.jar generate -i .\swagger.json -c .\config.json -l java -o labapi `

Where 
swagger.json ->https://msidlab.com/swagger/v1/swagger.json
config.json -> 
```
{
    "modelPackage": "com.microsoft.identity.internal.test.labapi.model",
    "apiPackage": "com.microsoft.identity.internal.test.labapi.api",
    "invokerPackage": "com.microsoft.identity.internal.test.labapi",
}
```

Additional Changes
1. Updated the generated ApiClient to use OAuth as the authentications (ApiClient.java: 96), This is to ensure that we can use accessToken as authentication to call labapis.
2. Updated the ApiNames as per the latest generated code

Verifications:
1. Verified running test locally to make sure we are able to call labapis and get lab accounts/secrets to use in automation
2. Automation runs in PR pipelines succeeded

### CODE REVIEW

**Comment by shahzaibj** [comment]
I am not sure what you mean by the following?

> Updated the generated ApiClient to use OAuth as the authentications (ApiClient.java: 96), This is to ensure that we can use accessToken as authentication to call labapis.

How was it working until today then? We were already using access token....so not clear to me what you changed here

**Review by shahzaibj — CHANGES_REQUESTED** [review]
Tests are failing

**Comment by iamgusain** [comment]
> I am not sure what you mean by the following?
> 
> > Updated the generated ApiClient to use OAuth as the authentications (ApiClient.java: 96), This is to ensure that we can use accessToken as authentication to call labapis.
> 
> How was it working until today then? We were already using access token....so not clear to me what you changed here

This is the additional change we need to do in the autogenerated code, the code to use access token doesn't get auto generated, We have to manually add it. I added it here in the PR description for future reference.

**Comment by iamgusain** [comment]
> Tests are failing

Yeah I see that some of the Msal tests are failing (all with same error). I will take a look
Good that Broker tests are passing

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabUserHelper.java [hunk 1]
```diff
@@ -233,8 +235,13 @@ public static String loadUserForTest(LabUserQuery query) {
...
 
         LabConfig.setCurrentLabConfig(labConfig);
+        UserInfo userInfo = labConfig.getConfigInfo().getUserInfo();
+        String upn = userInfo.getHomeUPN();
+        if (StringUtil.isNullOrEmpty(upn) || upn.equalsIgnoreCase("None")) {
+            upn = userInfo.getUpn();
+        }
 
-        return labConfig.getConfigInfo().getUserInfo().getHomeUPN();
+        return upn;
```
Comment (shahzaibj) [inline testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabUserHelper.java]: Has something changed that is causing us to make this change? Why was it working all the way until now?

Comment (iamgusain) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabUserHelper.java]: We need to use the HomeUpn property for cross-cloud scenarios

Comment (shahzaibj) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabUserHelper.java]: and looks like we were already using that, right? Now you've added a fallback to use the `upn` property if `homeUpn` is not there...what's the reason behind this?

Comment (iamgusain) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabUserHelper.java]: No, we are using upon currently.

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabDeviceHelper.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ public static boolean deleteDevice(final String upn, final String deviceId) thro
 
         try {
             final CustomSuccessResponse customSuccessResponse;
-            customSuccessResponse = deleteDeviceApi.delete(upn, deviceId);
+            customSuccessResponse = deleteDeviceApi.apiDeleteDeviceDelete(upn, deviceId);
```
Comment (shahzaibj) [inline testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabDeviceHelper.java:49]: these new names seem to be interesting

Comment (shahzaibj) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/LabDeviceHelper.java:49]: I am glad now that we have helpers for these..

**Review by shahzaibj — APPROVED** [review]
LGTM!

**Review by shoatman — APPROVED** [review]
I approved.  Some of the newly generated names look really strange... prefix with "api".. is this as a result of the swagger changes or something in the generation?

**Comment by iamgusain** [comment]
> I approved. Some of the newly generated names look really strange... prefix with "api".. is this as a result of the swagger changes or something in the generation?

Its the way the code geneartion library currently generates the code

**Pull request merged**